schema/sqlite/v3/temporal/schema.sql

```diff
@@ -93,6 +93,30 @@ CREATE TABLE task_queues (
 	PRIMARY KEY (range_hash, task_queue_id)
 );
 
+-- Stores activity or workflow tasks
+-- Used for fairness scheduling. (pass, task_id) are monotonically increasing.
+CREATE TABLE tasks_v2 (
+	range_hash INT UNSIGNED NOT NULL,
+	task_queue_id VARBINARY(272) NOT NULL,
+	pass BIGINT NOT NULL, -- pass for tasks (see stride scheduling algorithm for fairness)
+	task_id BIGINT NOT NULL,
+	--
+	data MEDIUMBLOB NOT NULL,
+	data_encoding VARCHAR(16) NOT NULL,
+	PRIMARY KEY (range_hash, task_queue_id, pass, task_id)
+);
+
+-- Stores ephemeral task queue information such as ack levels and expiry times
+CREATE TABLE task_queues_v2 (
```

Comment:
The separate task queue table is needed to ensure consistency for v2 tasks and the task queue data?

Reply:
Technically for SQL we could use the same task_queues table since we can do cross-table transactions. But it would get pretty confusing: we could combine the metadata for both tables in the same blob. Or we could use a prefix on task_queue_id (but then we might as well use another table).

Basically I think the main thing is: for cassandra we have to keep them separate (the fair tasks and fifo tasks are just separate queues entirely), so the code assumes that, so for sql we should do the same to avoid confusion.

Reply:
👍 makes sense